src/platforms/common/stm32/gdb_if.c

```diff
@@ -88,15 +88,20 @@ static void gdb_if_update_buf(void)
 	count_out = usbd_ep_read_packet(usbdev, CDCACM_GDB_ENDPOINT, buffer_out, CDCACM_PACKET_SIZE);
 	out_ptr = 0;
 #else
-	__asm__ volatile("cpsid i; isb");
+	cm_disable_interrupts();
+	__asm__ volatile("isb");
 	if (count_new) {
 		memcpy(buffer_out, double_buffer_out, count_new);
 		count_out = count_new;
 		count_new = 0;
 		out_ptr = 0;
 		usbd_ep_nak_set(usbdev, CDCACM_GDB_ENDPOINT, 0);
 	}
-	__asm__ volatile("cpsie i; isb");
+	cm_enable_interrupts();
+	__asm__ volatile("isb");
+	if (!count_new) {
```

Comment:
Okay, sure.
The meaning of `count_out` is not fully clear to me -- maybe the second WFI needs moving up into STM32F1-specific part of the function? I observed a non-zero count_out during inception debugging the blackpill-f4, and it didn't sleep once any gdb_if was open.

Reply:
It's not at all clear to us either - think this is probably a good call as a just in case, given the WFI'ing should probably be either-or (either `count_new` or `count_out`) gated